libraries/AP_BattMonitor/AP_BattMonitor_Backend.cpp

```diff
@@ -214,4 +217,47 @@ void AP_BattMonitor_Backend::check_failsafe_types(bool &low_voltage, bool &low_c
     } else {
         low_capacity = false;
     }
+
+    // check battery cell voltages
+    if (has_cell_voltages()) {
+
+        uint16_t cell_diff_mv = 0.0f;
+        uint16_t cell_lowest_mv = _state.cell_voltages.cells[0];  // initialize with the first cell's voltage
+        uint16_t cell_highest_mv = _state.cell_voltages.cells[0]; // initialize with the first cell's voltage
+
+        for (uint8_t i = 0; i < ARRAY_SIZE(_state.cell_voltages.cells); i++) {
+            
+            // bail out if cell voltage array element is UINT16_MAX since that means we've reached elements
+            if (_state.cell_voltages.cells[i] == UINT16_MAX) {
```

Comment:
This isn't actually true. You need to loop through the whole array. On Maxell back ends we can lose the middle voltage and fill it in as `UINT16_MAX` if that happens. (The Maxell behavior is admittedly a bit questionable, but that is the current state of implementation.)

Reply:
lol ok